src/Sticky/index.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const POSITION_VARIANTS = [
+  'top',
+  'bottom',
+];
+
+const Sticky = ({
+  position,
+  children,
+  offset,
+  className,
+  style,
```

Comment:
[clarification] Should `style` be explicitly handled by this component? Generally, using inline styles are avoided so I'd be make `style` be part of the official API for this component.

Instead, could `style` be replaced with `...rest` and update below by spreading on `...rest` like `<div {...rest} />`?

Reply:
right, `{...rest}` is more appropriate. `Style` was added as for the case when user would want to add an offset programmatically. And `{...rest}` allows this, cool.